### PR TITLE
Use custom JsonSerializer to serialize StreamlineConfiguration

### PR DESCRIPTION
**Pull request by raju-saravanan**
This commit contains following changes :
  * Move configurations and resources under webservice to their respective directories.
  * Instead of serializing StreamlineConfiguration manually abstract out serializing logic to its dedicated JsonSerializer.

### CODE REVIEW

**Comment by harshach** [comment]
@raju-saravanan what is this change about. Can you file a ISSUE and describe the changes there.

**Comment by raju-saravanan** [comment]
Details of the issue can be found here #820

File: webservice/src/main/java/com/hortonworks/streamline/webservice/StreamlineApplication.java [hunk 1]
```diff
@@ -41,6 +41,8 @@
 import com.hortonworks.streamline.streams.security.service.SecurityCatalogService;
 import com.hortonworks.streamline.streams.service.GenericExceptionMapper;
 
+import com.hortonworks.streamline.webservice.configurations.*;
```
Comment (HeartSaVioR) [inline webservice/src/main/java/com/hortonworks/streamline/webservice/StreamlineApplication.java]: Let's avoid having wildcard import.

File: webservice/src/main/java/com/hortonworks/streamline/webservice/util/StreamlineConfigurationSerializer.java [hunk 2]
```diff
@@ -1,57 +1,32 @@
-/**
-  * Copyright 2017 Hortonworks.
-  *
-  * Licensed under the Apache License, Version 2.0 (the "License");
-  * you may not use this file except in compliance with the License.
-  * You may obtain a copy of the License at
-
-  *   http://www.apache.org/licenses/LICENSE-2.0
+package com.hortonworks.streamline.webservice.util;
```
Comment (HeartSaVioR) [inline webservice/src/main/java/com/hortonworks/streamline/webservice/util/StreamlineConfigurationSerializer.java]: Please add license header above this line.

File: webservice/src/main/java/com/hortonworks/streamline/webservice/StreamlineApplication.java (hunk 1 above)
Comment (raju-saravanan) [inline webservice/src/main/java/com/hortonworks/streamline/webservice/StreamlineApplication.java]: Removed wildcard import

File: webservice/src/main/java/com/hortonworks/streamline/webservice/util/StreamlineConfigurationSerializer.java (hunk 2 above)
Comment (raju-saravanan) [inline webservice/src/main/java/com/hortonworks/streamline/webservice/util/StreamlineConfigurationSerializer.java]: I have added the license header for this file

**Comment by raju-saravanan** [comment]
@HeartSaVioR  can you please review this?

**Comment by HeartSaVioR** [comment]
I noticed that it always build a map for every requests whereas it didn't. I think the endpoint will be called occasionally or even rarely so that's OK to me, but we may even create serialized config instance once and reuse it if we don't plan to support dynamic configuration.

+1
@raju-saravanan Thanks for the patch.

**Comment by raju-saravanan** [comment]
Thanks @HeartSaVioR . I agree, if this becomes a bottleneck in the future, we can store serialized config once and reuse it with subsequent calls to the API.

**Comment by raju-saravanan** [comment]
@HeartSaVioR can you please merge this patch to the master?

**Comment by HeartSaVioR** [comment]
@raju-saravanan Sorry to revisit late. This now needs resolving conflicts. Please rebase and let me know. After that I'll merge immediately.

**Comment by raju-saravanan** [comment]
@HeartSaVioR I have rebased my changes with the master

**Comment by HeartSaVioR** [comment]
This closes #820

**Pull request merged**